core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/Chars.java

```diff
@@ -0,0 +1,30 @@
+package org.eclipse.rdf4j.rio.turtle;
+
+/**
+ * Utilities that were taken from the Character but changed to be better suited for parsing
+ * without additional garbage.
+ * @author Nikola Petrov nikola.petrov@ontotext.com
+ */
+class Chars {
+	/**
+	 * Appends the characters from codepoint into the string builder. This is the same
+	 * as Character#toChars but prevents the additional char array garbage.
+	 * @param dst the destination in which to append the characters
+	 * @param codePoint the codepoint to be appended
+	 */
+	public static void append(StringBuilder dst, int codePoint) {
```

Comment:
The snippet was taken from OpenJDK, I guess there is no problem with it as long as the result is open source again. 

I put the class as package private and created a dedicated class so we can update it easier in the future(if there is a need). But OK will move it in TurtleParser.